### PR TITLE
Break out the CMS related code into a new class.

### PR DESCRIPTION
**Pull request by Martin2112**
Break out the CMS related code (v2 related) into a new class and move tests out of cert_checker_test.

### CODE REVIEW

**Comment by Martin2112** [comment]
This is almost identical to code submitted last week. Just pulls it out and tweaks the API.

File: cpp/log/cert_verifier.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#include <openssl/cms.h>
+
+#include "base/macros.h"
+#include "log/cert.h"
+#include "util/openssl_util.h"  // for LOG_OPENSSL_ERRORS
+#include "util/status.h"
+
+namespace cert_trans {
+
+class CertVerifier {
```
Comment (AlCutter) [inline cpp/log/cert_verifier.h]: Having `CertChecker` and `CertVerifier` could get a bit confusing, how about `CMSVerifier`, (although maybe that's not so good as this doesn't [currently] verify the signature)?

**Comment by Martin2112** [comment]
Naming things is hard! I would maybe go with CMSVerifier. It does verify the CMS signature of the message, but not that the signer cert is part of a valid chain as we use in CT.

File: cpp/log/cert.h [hunk 2]
```diff
@@ -182,6 +182,8 @@ class Cert {
 
   // CertChecker needs access to the x509_ structure directly.
   friend class CertChecker;
+  // CertVerifier needs access to the x509_ structure directly.
```
Comment (AlCutter) [inline cpp/log/cert.h]: `CmsVerifier needs ...` :)

File: cpp/log/cms_verifier.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
+/* -*- indent-tabs-mode: nil -*- */
+#include <log/cms_verifier.h>
```
Comment (AlCutter) [inline cpp/log/cms_verifier.cc]: `#include "log/cms_verifier.h"`

File: cpp/log/cert.h (hunk 2 above)
Comment (Martin2112) [inline cpp/log/cert.h]: So Eclipse refactor didn't work? The world is back to normal now. Will fix it.

File: cpp/log/cms_verifier_test.cc [hunk 4]
```diff
@@ -0,0 +1,211 @@
+/* -*- indent-tabs-mode: nil -*- */
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include <log/cms_verifier.h>
```
Comment (AlCutter) [inline cpp/log/cms_verifier_test.cc]: `"log/cms_verifier.h"` and move down to block below

File: cpp/log/cms_verifier_test.cc [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+// issuer of the embedded cert
+static const char kCmsSignedDataTest5[] = "cms_test5.der";
+
+// Subject name we expect in our embedded certificate CMS tests
+static const char kCmsTestSubject[] = "CN=?.example.com, C=GB, ST=Wales, "
+    "L=Erw Wen, O=Certificate Transparency";
+
+namespace {
+
+class CertVerifierTest : public ::testing::Test {
```
Comment (AlCutter) [inline cpp/log/cms_verifier_test.cc]: CmsVerifierTest

File: cpp/log/cert.h (hunk 2 above)
Comment (Martin2112) [inline cpp/log/cert.h]: Done.

File: cpp/log/cms_verifier.cc (hunk 3 above)
Comment (Martin2112) [inline cpp/log/cms_verifier.cc]: Done.

File: cpp/log/cms_verifier_test.cc (hunk 4 above)
Comment (Martin2112) [inline cpp/log/cms_verifier_test.cc]: Done.

File: cpp/log/cms_verifier_test.cc (hunk 5 above)
Comment (Martin2112) [inline cpp/log/cms_verifier_test.cc]: Done + other occurrences.

**Comment by Martin2112** [comment]
I think that's all fixed up now.

File: cpp/log/cms_verifier_test.cc [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+  void SetUp() {
+    CHECK(util::ReadTextFile(cert_dir_ + "/" + kLeafCert, &leaf_pem_))
+        << "Could not read test data from " << cert_dir_
+        << ". Wrong --test_srcdir?";
+    CHECK(util::ReadTextFile(cert_dir_ + "/" + kIntermediateCert,
+                             &intermediate_pem_));
+    CHECK(util::ReadTextFile(cert_dir_ + "/" + kCaCert, &ca_pem_));
+  }
+};
+
```
Comment (AlCutter) [inline cpp/log/cms_verifier_test.cc]: Last thing: could you pop two lines between top-level items here & elsewhere?

Comment (Martin2112) [inline cpp/log/cms_verifier_test.cc]: Done. But I'm not sure it's a great idea as it can't be done automatically by formatters.

**Pull request merged**